mesonbuild/interpreter/interpreter.py

```diff
@@ -1930,6 +1934,23 @@ def func_install_emptydir(self, node: mparser.BaseNode, args: T.Tuple[str], kwar
 
         return d
 
+    @FeatureNew('install_symlink', '0.61.0')
+    @typed_kwargs(
+        'install_symlink',
+        KwargInfo('pointing_to', str),
+        KwargInfo('install_dir', (str, NoneType)),
+        KwargInfo('install_tag', (str, NoneType)),
+    )
+    def func_install_symlink(self, node: mparser.BaseNode,
+                             args: T.Tuple[T.List[str]],
+                             kwargs) -> build.SymlinkData:
+        name = args[0]
```

Comment:
This should check that `len(args) == 1` and that `args[0]` is a string and does not contain `/` or `\`.

There should also be a `failing` test that tries to install a symlink called `foo/bar` which then fails.

Reply:
We can use `typed_pos_args('install_symlink', str)` for this. It won't validate that it doesn't contain path separators though.

Reply:
Thank for the suggestions! Adding the test found a bug, so that was definitely a requirement... 

I used the positional argument. The path separator verification is done inside `build.SymlinkData`. I added a comment so that it is more obvious.